bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/discovery/TouchWandControllerDiscoveryService.java

```diff
@@ -0,0 +1,145 @@
+
+    private static final int SEARCH_TIME = 2;
+    private static final int TOUCHWAND_BCAST_PORT = 35000;
+    private Thread socketReceiveThread = null;
+    private DatagramSocket listenSocket = null;
+
+    private final Logger logger = LoggerFactory.getLogger(TouchWandControllerDiscoveryService.class);
+
+    public TouchWandControllerDiscoveryService() {
+        super(TouchWandBridgeHandler.SUPPORTED_THING_TYPES, SEARCH_TIME, true);
+        removeOlderResults(getTimestampOfLastScan());
+    }
+
+    @Override
+    protected void startScan() {
+        try {
+            listenSocket = new DatagramSocket(TOUCHWAND_BCAST_PORT);
+        } catch (SocketException e) {
+            logger.warn("SocketException {}", e.getMessage());
+        }
+        runReceiveThread(listenSocket);
+    }
+
+    @Override
+    protected synchronized void stopScan() {
+        super.stopScan();
+    }
+
+    @Override
+    public void deactivate() {
+        super.deactivate();
+        if (socketReceiveThread != null) {
+            socketReceiveThread.interrupt();
+            socketReceiveThread = null;
+        }
+        if (listenSocket != null) {
+            listenSocket.close();
+        }
+    }
+
+    private void addDeviceDiscoveryResult(String label, String ip) {
+        String id = ip.replaceAll("\\.", "");
+        ThingUID thingUID = new ThingUID(THING_TYPE_BRIDGE, id);
+        Map<String, Object> properties = new HashMap<>();
+        properties.put("label", label);
+        properties.put("ipAddress", ip);
+        // @formatter:off
+        logger.debug("Add new Bridge label:{} id {} ",label, id);
+        thingDiscovered(DiscoveryResultBuilder.create(thingUID)
+                .withThingType(THING_TYPE_BRIDGE)
+                .withLabel(label)
+                .withProperties(properties)
+                .withRepresentationProperty(ip)
+                .build()
+        );
+        // @formatter:on
+    }
+
+    protected void runReceiveThread(DatagramSocket socket) {
+        socketReceiveThread = new ReceiverThread(socket);
+        socketReceiveThread.start();
+        socketReceiveThread.setName(TouchWandBindingConstants.BINDING_ID);
+    }
+
+    @NonNullByDefault
+    private class ReceiverThread extends Thread {
+
+        private static final int BUFFER_LENGTH = 256;
+        private DatagramPacket dgram = new DatagramPacket(new byte[BUFFER_LENGTH], BUFFER_LENGTH);
+        DatagramSocket mySocket;
+
+        public ReceiverThread(DatagramSocket socket) {
+            mySocket = socket;
+        }
+
+        @Override
+        public void run() {
+            receiveData(dgram);
+        }
+
+        private void receiveData(DatagramPacket datagram) {
+            try {
+                while (true) {
+                    mySocket.receive(datagram);
+                    InetAddress address = datagram.getAddress();
+                    String sentence = new String(dgram.getData(), 0, dgram.getLength(), "US-ASCII");

```

Comment:
Using a constant will prevent throwing an UnsupportedEncodingException:
```suggestion
                    String sentence = new String(dgram.getData(), 0, dgram.getLength(), StandardCharsets.US_ASCII);
```